### PR TITLE
[Videos] handle app backgrounding

### PR DESCRIPTION
**Pull request by mozzius**
1. Handles app backgrounding
2. Ensures video plays when fullscreening and unfullscreening, just in case it manages to get paused somehow

### CODE REVIEW

File: src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx [hunk 1]
```diff
@@ -101,6 +114,24 @@ function Controls({
...
+      case 'idle':
+      case 'loading':
+      case 'readyToPlay': {
+        if (!player.playing) player.play()
+        enterFullscreen()
+        break
+      }
+      case 'error': {
+        // not exactly sure what to do here
+        // maybe replay the video?
```
Comment (haileyok) [inline src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx:128]: should be okay, can rm comment if you want

File: src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx [hunk 2]
```diff
@@ -21,6 +22,18 @@ export function VideoEmbedInnerNative() {
...
+  const [prevFocused, setPrevFocused] = useState(isAppFocused)
+
+  // resume video when coming back from background
+  useEffect(() => {
+    if (isAppFocused !== prevFocused) {
+      setPrevFocused(isAppFocused)
+      if (isAppFocused === 'active') {
+        player.play()
+      }
+    }
```
Comment (haileyok) [inline src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx:35]: Might be wrong, but is there a reason that we need to store the previous state? Seems that we might be able to just do

```tsx
const appState = useAppState()

useEffect(() => {
  if (appState !== 'active') return
  player.play()
}, [appState, player])
```

Seems that `player` should be stable so the only thing that should change here is `appState`.

**Review by haileyok — APPROVED** [review]
One note but lgtm.

File: src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx (hunk 2 above)
Comment (mozzius) [reply src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx:35]: I wanted to constrain when it fires specifically to when it's going from !active -> active. Your version will also fire on initial render. idk, do you think that matters?

Comment (haileyok) [reply src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx:35]: Ahh, right. Maybe let's just stick the prev value in a ref then?

Comment (mozzius) [reply src/view/com/util/post-embeds/VideoEmbedInner/VideoEmbedInnerNative.tsx:35]: Yeah that'd work

**Pull request merged**